### PR TITLE
feat(nodes): add invocation schema customisation, add model selection

### PR DESCRIPTION
**Pull request by psychedelicious**
- add invocation schema customisation

done via fastapi's `Config` class and `schema_extra`. when using `Config`, inherit from `InvocationConfig` to get type hints.

where it makes sense - like for all math invocations - define a `MathInvocationConfig` class and have all invocations inherit from it.

this customisation can provide any arbitrary additional data to the UI. currently it provides tags and field type hints.

this is necessary for `model` type fields, which are actually string fields. without something like this, we can't reliably differentiate  `model` fields from normal `string` fields.

can also be used for future field types.

all invocations now have tags, and all `model` fields have ui type hints.

- fix model handling for invocations

added a helper to fall back to the default model if an invalid model name is chosen. model names in graphs now work.

- fix latents progress callback

noticed this wasn't correct while working on everything else.

### CODE REVIEW

**Review by Kyle0654 — APPROVED** [review]
It's too bad we can't get config classes to merge info from multiple base classes. It would be a lot cleaner if we could just defined tags/ui hints on base classes and pull those in.

File: invokeai/app/invocations/generate.py [hunk 1]
```diff
@@ -6,22 +6,40 @@
...
+class SDImageInvocation(BaseModel):
+    """Helper class to provide all Stable Diffusion raster image invocations with additional config"""
+
+    # Schema customisation
+    class Config(InvocationConfig):
+        schema_extra = {
+            "ui": {
+                "tags": ["stable-diffusion", "image"],
+                "type_hints": {
+                    "model": "model",
```
Comment (Kyle0654) [inline invokeai/app/invocations/generate.py:35]: The one worry I have with this is that it's pretty hidden/separate from the fields below. Someone adding a `model` field to some other invocation may not realize they need to also be adding this to their config.

**Comment by psychedelicious** [comment]
> It's too bad we can't get config classes to merge info from multiple base classes. It would be a lot cleaner if we could just defined tags/ui hints on base classes and pull those in.

I think there's a way to do this, maybe with (metaclasses)[https://docs.python.org/3/reference/datamodel.html#metaclasses], or just cleverly using normal classes, but Im not well-versed in OOP and the python syntax is difficult to reason about without types.

Another option is (decorators with parameters)[https://www.artima.com/weblogs/viewpost.jsp?thread=240845#decorator-functions-with-decorator-arguments]:

```python
@tags(["image", "latents"])
@type_hints({ "model": "model" })
class LatentToImageInvocation(BaseModel):
```

Still you'd need to handle building the Config class (merging the tags and type_hints into the Config) in the decorator and giving it to the Invocation. I don't understand the OOP stuff well enough to implement this.

File: invokeai/app/invocations/generate.py (hunk 1 above)
Comment (psychedelicious) [reply invokeai/app/invocations/generate.py:35]: Yeah I agree. For now, this works, but maybe a python person can figure out a better solution. The main thing this PR introduces is the structure of the UI hints, with the implementation kinda just being what I could figure out.

**Pull request merged**